mesonbuild/arglist.py

```diff
@@ -264,8 +264,8 @@ def extend_direct(self, iterable: T.Iterable[str]) -> None:
             self.append_direct(elem)
 
     def extend_preserving_lflags(self, iterable: T.Iterable[str]) -> None:
-        normal_flags = []
-        lflags = []
+        normal_flags: T.List[str] = []
+        lflags: T.List[str] = []
```

Comment:
> overwhelmingly this commit just adds explicit type annotations for variables. This assists type checkers and can help the programmer know if that added an int to what was an implicit list of strings, for instance.

mypy actually infers this as T.List[str] since the first action performed on it is to append something of type str. So appending an int would be a type error, but if the first append was for an int, then appending a str would be a type error.

e.g. if you try adding `.append(1)` at the end of the function, it throws:
> Argument 1 to "append" of "list" has incompatible type "int"; expected "str"  [arg-type]

If it cannot figure out how to properly type narrow on the first usage, then does demand you explicitly annotate it:
> Need type annotation for "mylist" (hint: "mylist: List[<type>] = ...")  [var-annotated]

Reply:
This commit is really an aid for VSCode users. Though mypy seems to be happy with most of these types, pylance/pyright need a little aid, and I find the readability a little bit easier. Is there anything actionable you want me to do here?